core/string/ustring.cpp

```diff
@@ -4305,6 +4305,64 @@ String String::replace_first(const char *p_key, const char *p_with) const {
 	return *this;
 }
 
+String String::replace_char(char32_t p_key, char32_t p_with) const {
+	ERR_FAIL_COND_V_MSG(p_with == 0, String(), "`with` must not be null.");
+
+	if (p_key == 0 || is_empty()) {
+		return *this;
+	}
+
+	String new_string;
+
+	new_string.resize(size());
+
+	char32_t *new_ptrw = new_string.ptrw();
```

Comment:
Copying the string is risky performance wise! `_replace_common` does a full round to check if any adjustments are needed, and if not, returns the string unchanged. It's possible that this implementation is slower than the string-replace if no changes are needed.
In my own PR, I avoided this by only forking `_copy_on_write` if actually needed :)

Reply:
Will investigate, would be trivial to perform that alteration by having a first pass identifying the first position if any